cfgov/jinja2/v1/_layouts/base.html

```diff
@@ -241,48 +247,58 @@
 #}
 
 {% block javascript scoped %}
-<!--[if gt IE 8]><!-->
-{# Include site-wide JavaScript. #}
-<script src="{{ static('js/routes/common.js') }}"></script>
-
-{# Check and include template-level JavaScript. #}
-{% macro template_js() %}
-    {% include 'js/routes/' + request.path.split('/')[1] + '/single.js' ignore missing %}
-{% endmacro %}
-{% set js_source = template_js() | trim %}
-{% if js_source | length > 0 %}
-<script src="{{ static('js/routes/' + request.path.split('/')[1] + '/single.js') }}"></script>
-{% endif %}
 
-{# Check and include component-level JavaScript. #}
-{% macro component_ondemand_js(js) %}
-    {% include 'js/routes/on-demand/' + js ignore missing %}
-{% endmacro %}
+{#
+  Modernizr adds a no-js class the HTML element if ECMAScript 5 features aren't
+  available. Check for that CSS class here and act accordingly.
+#}
+<script>
+  if ( document.body.parentElement.className.indexOf( 'no-js' ) === -1 ) {
 
-{% if page and page.media %}
-    {% compress js %}
-    <script>
-    {% for js in page.media %}
-       {% set component_js_source = component_ondemand_js(js) | trim %}
-       {% if component_js_source | length > 0 %}
-            {{ component_js_source }}
-       {% endif %}
-    {% endfor %}
-    </script>
-    {% endcompress %}
-{% endif %}
+    {# Include site-wide JavaScript. #}
+    var script = document.createElement( 'script' );
+      script.type = 'text/javascript';
+      script.src = '{{ static('js/routes/common.js') }}';
+      document.getElementsByTagName( 'head' )[0].appendChild( script );
+
+    {# Check and include template-level JavaScript. #}
+    {% macro template_js() %}
+       {% include 'js/routes/' + request.path.split('/')[1] + '/single.js' ignore missing %}
+    {% endmacro %}
+    {% set js_source = template_js() | trim %}
+    {% if js_source | length > 0 %}
+      script = document.createElement( 'script' );
+      script.type = 'text/javascript';
+      script.src = '{{ static('js/routes/' + request.path.split('/')[1] + '/single.js') }}';
+      document.getElementsByTagName( 'head' )[0].appendChild( script );
+    {% endif %}
 
-{# TODO: Remove the below block when all pages
-         using this template are moved to Wagtail. #}
-{# Check and include page-level JavaScript. #}
-{% macro page_js() %}
-    {% include 'js/routes/' + request.path[1:] + 'index.js' ignore missing %}
-{% endmacro %}
-{% set js_source = page_js() | trim %}
-{% if js_source | length > 0 %}
-<script src="{{ static('js/routes/' + request.path[1:] + 'index.js') }}"></script>
-{% endif %}
-<!--<![endif]-->
+    {# Check and include component-level JavaScript. #}
+    {% if page and page.media %}
+       {% for js in page.media %}
+         script = document.createElement( 'script' );
```

Comment:
would it make sense to create a small inline function so you don't have to repeat the create / append/ src code. I think that is what they are doing for the GTM stuff.

Reply:
We have https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/unprocessed/js/modules/util/js-loader.js. We could copy some of that code over.

Reply:
that works, but minimize the code when you add it base.html ( you can just link to the source ).

Reply:
Added it here https://github.com/cfpb/cfgov-refresh/pull/3677/files#diff-89b43ee86bbd4952dc44cf751cfd775cR256